tests/unit/test_charm.py

```diff
@@ -60,14 +70,12 @@ def test_on_config_changed(self, _check_pgb_running, _mkdir, _reload, _update_co
                 "listen_port": 6464,
             }
         )
-        _reload.assert_called()
-        _update_connection_info.assert_called()
-        _check_pgb_running.assert_called()
-
-        # Test changing charm config propagates to container config file.
-        pgb_container = self.harness.model.unit.get_container(PGB)
-        container_config = pgb_container.pull(INI_PATH).read()
-        self.assertEqual(container_config, test_config.render())
+        _reload.assert_called_once_with()
+        _update_connection_info.assert_called_with(6464)
+        _check_pgb_running.assert_called_once_with()
+        _push_file.assert_called_with(
+            "/var/lib/pgbouncer/pgbouncer.ini", test_config.render(), 0o400
+        )
```

Comment:
Ops 2.5 changed the way testing of container fs works, so I redid the test to sidestep the changes.